### PR TITLE
Darwinia Spec 1101

### PR DESCRIPTION
**Pull request by AurevoirXavier**
Migrate `DarwiniaEthereumBacking` to `EthereumBacking`

<sup> Created from VS Code using [CodeStream](https://codestream.com/?utm_source=cs&utm_medium=pr&utm_campaign=github*com)</sup>

### CODE REVIEW

File: runtime/darwinia/src/lib.rs [hunk 1]
```diff
@@ -575,48 +575,46 @@ pub struct CustomOnRuntimeUpgrade;
...
 		RuntimeBlockWeights::get().max_block
 	}
 }
+
+fn migrate() {
+	// --- substrate ---
+	use frame_support::migration;
+
+	log::info!("Moving storage `DarwiniaEthereumBacking` to `EthereumBacking`");
+	migration::move_pallet(b"DarwiniaEthereumBacking", b"EthereumBacking");
```
Comment (HackFisher) [inline runtime/darwinia/src/lib.rs:619]: The event storage key in Ethereum smart contract for locking assets need be upgrade too. @hujw77 @WoeOm

**Comment by AurevoirXavier** [comment]
@wi1dcard 

Is this name available?

```
docker pull quay.io/darwinia-network/darwinia:v0.11.0-fix-storage
```

**Comment by wi1dcard** [comment]
> @wi1dcard
> 
> Is this name available?
> 
> ```
> docker pull quay.io/darwinia-network/darwinia:v0.11.0-fix-storage
> ```

Yes, it conforms to the name convention of docker images.

**Comment by AurevoirXavier** [comment]
> > @wi1dcard
> > Is this name available?
> > ```
> > docker pull quay.io/darwinia-network/darwinia:v0.11.0-fix-storage
> > ```
> 
> Yes, it conforms to the name convention of docker images.

Where can I specify the name? In `toml`, `README` or somewhere.

**Comment by wi1dcard** [comment]
> Where can I specify the name? In `toml`, `README` or somewhere.

Just create a tag named `v0.11.0-fix-storage`. Or if you'd like to have a tag with a different name, please let me know after the tag has been created and I will manually push the docker images.

**Comment by AurevoirXavier** [comment]
> Just create a tag named `v0.11.0-fix-storage`. Or if you'd like to have a tag with a different name, please let me know after the tag has been created and I will manually push the docker images.

Great, I'll create a tag named `v0.11.0-fix-storage` directly.

**Comment by HackFisher** [comment]
If between 1100 and 1101, some one write some storage in the same value or map key, will the migrate overwrite. Will it abort due to there are already exist?

**Comment by AurevoirXavier** [comment]
Try Runtime

<details>
<summary>Click to expand!</summary>

```
2021-07-29 17:41:01 `pallet_timestamp::UnixTime::now` is called at genesis, invalid value returned: 0
2021-07-29 17:41:01 [#0] 🗳  Finalized election round with compute ElectionCompute::OnChain.
2021-07-29 17:41:01 [0] 💸 new validator set of size 1 has been processed for era 1
2021-07-29 17:41:01 initializing remote client to "http://localhost:20000"
2021-07-29 17:41:01 scraping keypairs from remote node http://localhost:20000 @ 0x59603ed330fa54b728809fe130ee783dbcff169c91d27fc31c29c85b59d79136
2021-07-29 17:41:01 downloading data for all modules.
2021-07-29 17:41:36 extending externalities with 1 manually injected keys
2021-07-29 17:41:36 injecting a total of 98188 keys
2021-07-29 17:41:38 try-runtime::on_runtime_upgrade Darwinia.
2021-07-29 17:41:38 Moving storage `DarwiniaEthereumBacking` to `EthereumBacking`
2021-07-29 17:41:38 ⚠️ System declares internal migrations (which *might* execute), setting storage version to PalletVersion { major: 3, minor: 0, patch: 0 }
2021-07-29 17:41:38 Moving storage `DarwiniaEthereumBacking` to `EthereumBacking`
2021-07-29 17:41:38 ✅ no migration for TronBacking, setting storage version to PalletVersion { major: 2, minor: 3, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for EthereumRelayAuthorities, setting storage version to PalletVersion { major: 2, minor: 3, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for EthereumRelayerGame, setting storage version to PalletVersion { major: 2, minor: 3, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for EthereumBacking, setting storage version to PalletVersion { major: 2, minor: 3, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for EthereumRelay, setting storage version to PalletVersion { major: 2, minor: 3, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for CrabBacking, setting storage version to PalletVersion { major: 2, minor: 3, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for Multisig, setting storage version to PalletVersion { major: 3, minor: 0, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for Proxy, setting storage version to PalletVersion { major: 3, minor: 0, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for Scheduler, setting storage version to PalletVersion { major: 3, minor: 0, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for Recovery, setting storage version to PalletVersion { major: 3, minor: 0, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for Society, setting storage version to PalletVersion { major: 3, minor: 0, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for Identity, setting storage version to PalletVersion { major: 3, minor: 0, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for Utility, setting storage version to PalletVersion { major: 3, minor: 0, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for Vesting, setting storage version to PalletVersion { major: 2, minor: 3, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for Sudo, setting storage version to PalletVersion { major: 3, minor: 0, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for Treasury, setting storage version to PalletVersion { major: 2, minor: 3, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for TechnicalMembership, setting storage version to PalletVersion { major: 3, minor: 0, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for PhragmenElection, setting storage version to PalletVersion { major: 2, minor: 3, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for TechnicalCommittee, setting storage version to PalletVersion { major: 3, minor: 0, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for Council, setting storage version to PalletVersion { major: 3, minor: 0, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for Democracy, setting storage version to PalletVersion { major: 2, minor: 3, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for DarwiniaHeaderMMR, setting storage version to PalletVersion { major: 2, minor: 3, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for AuthorityDiscovery, setting storage version to PalletVersion { major: 3, minor: 0, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for ImOnline, setting storage version to PalletVersion { major: 3, minor: 0, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for Grandpa, setting storage version to PalletVersion { major: 3, minor: 0, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for Session, setting storage version to PalletVersion { major: 3, minor: 0, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for Historical, setting storage version to PalletVersion { major: 3, minor: 0, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for Offences, setting storage version to PalletVersion { major: 3, minor: 0, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for Staking, setting storage version to PalletVersion { major: 2, minor: 3, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for ElectionProviderMultiPhase, setting storage version to PalletVersion { major: 3, minor: 0, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for Authorship, setting storage version to PalletVersion { major: 3, minor: 0, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for TransactionPayment, setting storage version to PalletVersion { major: 3, minor: 0, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for Kton, setting storage version to PalletVersion { major: 2, minor: 3, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for Balances, setting storage version to PalletVersion { major: 2, minor: 3, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for Timestamp, setting storage version to PalletVersion { major: 3, minor: 0, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for Babe, setting storage version to PalletVersion { major: 3, minor: 0, patch: 0 }
2021-07-29 17:41:38 ✅ no migration for RandomnessCollectiveFlip, setting storage version to PalletVersion { major: 3, minor: 0, patch: 0 }
2021-07-29 17:41:38 try-runtime executed without errors. Consumed weight = 512972800000000, total weight = 512000000000000 (1.0019)
```
</details>

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,135 @@
...
+#### Fix(es)
+- None
+#### Migration(s)
+- None
+
+---
+
+#### Darwinia Custom
+##### Runtime
+- None
```
Comment (HackFisher) [inline CHANGELOG.md:52]:
```suggestion
- Although this version is released after 1110, but it will be upgraded first before 1110
```

**Comment by AurevoirXavier** [comment]
> If between 1100 and 1101, some one send transaction and write some storage in the same value or map key, will the migrate overwrite. Will it abort due to there are already exist?

Yes. I will be overwritten. But I didn't find any extrinsic could execute successfully.

**Comment by HackFisher** [comment]
https://darwinia.subscan.io/extrinsic/4352896-1

**Pull request merged**